cypress/e2e/bookStore.cy.js

```diff
@@ -1,11 +1,50 @@
 /// <reference types='cypress' />
 
 describe('Book Store app', () => {
-  before(() => {
+  // before(() => {
+  //   cy.visit('/login')
+  // });
+  const user = {
+    username: 'tester_1',
+    password: 'Tester1!'
+  };
 
+  const book = {
+    isbn: '9781449365035',
+    userId: 'a66cc17b-e2fc-40a5-920c-f6a9115834f9',
+    title: 'Speaking JavaScript'
+  };
+
+  it('should provide an ability to log in', () => {
+    cy.visit('/login');
+    cy.findByPlaceholder('UserName').type(user.username);
+    cy.findByPlaceholder('Password').type(user.password);
+    cy.get('button').contains('Login').click();
+    cy.get('label').contains(user.username).should('be.visible');
+    cy.get('button').contains('Log out').should('be.visible');
   });
 
-  it('', () => {
+  it('should provide an ability to add book to the cart', () => {
+    cy.visit('/login');
```

Comment:
There is no need to go to the login page if you log in via a request.

Reply:
fixed